jwk.go

```diff
@@ -401,6 +401,8 @@ func (k *JSONWebKey) Thumbprint(hash crypto.Hash) ([]byte, error) {
 		input, err = rsaThumbprintInput(key.N, key.E)
 	case ed25519.PrivateKey:
 		input, err = edThumbprintInput(ed25519.PublicKey(key[32:]))
+	case OpaqueSigner:
+		return key.Public().Thumbprint(hash)
```

Comment:
Thumbprint returns a slice of bytes so that makes sense to return early, but it also can return an error that that's not being handled yet.

Reply:
@pgporada Hi, but this actually return the tuple ([]byte, error). The call `key.Public()` returns a `*JSONWebKey`, and `*JSONWebKey.Thumbprint(hash)` is this method, so it's a recursive call.

I also added this PR https://github.com/go-jose/go-jose/pull/17 that I can work around. I know it's not in the standard, but it is Microsoft Azure, not something small.

Reply:
Ah, you're right. I didn't catch that the first time through.